### ISSUE TITLE
[Documentation] Improve documentation for Datatables / click:row

### ISSUE THREAD

**Issue by michaelisvy** [vuetifyjs/vuetify#9174]
### Problem to solve
hi,
first of all thanks a lot for the great work on Vuetify!
I am trying to do a simple datatable: it basically contains a list of flights and I would like to be able to click on one flight in order to display the flight details.
When I click on one row, I would like the datatable to disappear (so I was trying not to use slots).
I've gone multiple times through the documentation and didn't find out how to do it. 
The documentation for `click:row` says that `The item for the row is included`. I've been trying to access it in my code and didn't manage to have access to the clicked item. It would be great if the documentation could provide more detailed information. Here is what I've done:

```
<template>
  <v-data-table
    :headers="flightList.headers"
    :items="flightList.flights"
    item-key="id"
    @click:row="submitForm(item)"
  />
</template>
...

<script>
export default {
  props: {
    flightList: {
      required:true,
      type:Object
    }
  },
  data: () => {
    return {
    };
  },
  methods: {
    submitForm(item) {
      alert("hello!!! " + item);
    }
  }
};
</script>
```

### Proposed solution
A quick fix would be to make the documentation more explicit on how to retrieve the item after the corresponding row has been clicked. 
A more detailed way would be to provide an example using `@click:row` (I don't think there is any currently, unless I've missed it).

Great thanks!

**Comment by MajesticPotatoe** [comment]
`@click:row="submitForm"`
item is included means just that, its already included, thus doesn't need to be passed in manually

**Comment by michaelisvy** [comment]
thanks a lot! I've changed my code and it works now.
I've posted the updated snippet below. 

```
<template>
  <v-data-table
    :headers="flightList.headers"
    :items="flightList.flights"
    item-key="id"
    @click:row="submitForm"
  />
</template>
...

<script>
export default {
  props: {
    flightList: {
      required:true,
      type:Object
    }
  },
  data: () => {
    return {
    };
  },
  methods: {
    submitForm(item) {
      alert("hello!!! " + item);
    }
  }
};
</script>
```

**Comment by michaelisvy** [comment]
hi @MajesticPotatoe just wondering if you think we should update the documentation based on this issue? I'm happy to try and do a pull request if that could be of any help. 
If you think the doc is good enough, happy to close that issue as well.

**Comment by MajesticPotatoe** [comment]
@michaelisvy personally I think its good enough, though I'm not sure what more there would be to add without it being superfluous. That being said, if you have any suggestions I am more than happy to update it.

**Comment by michaelisvy** [comment]
I had another look at it. To me, what would be the most helpful would be to add one more example in the `datatables` page (unless you think it would clutter the page to have too many examples). 
As of now there is no example showcasing how to click on a row. 
I'm clearly a newbie in frontend Javascript, but it took me a few hours to understand how to do it (and that was all thanks to your help). It might help people like me if there is a bit more guidance :).

**Comment by MajesticPotatoe** [comment]
a simple example should suffice, will likely wait a little as data-table pages are due for a large update (there's a bunch of components and api missing) and we are in the process of restructuring docs. I'll leave this open for the time being.

**Comment by KaelWD** [comment]
Fixed by 6058523ccba46345838ec57a293f7e70716125b2
